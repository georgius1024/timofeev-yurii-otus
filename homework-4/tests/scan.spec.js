const scan = require('../src/scan')
describe('scanning module', async () => {
  it('can scan existing folder', async () => {
    const listing = await scan('./tests/foo')
    expect(listing).toMatchSnapshot()
  })

  it('can not scan non-existing folder', async () => {
    try {
      const listing = await scan('==error==')
    } catch (error) {
      expect(error.message).toEqual(expect.stringContaining('ENOENT'))
    }
  })
})
