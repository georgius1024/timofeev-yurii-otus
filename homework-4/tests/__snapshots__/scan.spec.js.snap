// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scanning module can scan existing folder 1`] = `
Object {
  "dirs": Array [
    "tests/foo",
    "tests/foo/bar",
    "tests/foo/bar/baz",
  ],
  "files": Array [
    "tests/foo/f1.txt",
    "tests/foo/f2.txt",
    "tests/foo/bar/bar1.txt",
    "tests/foo/bar/bar2.txt",
    "tests/foo/bar/baz/.gitignore",
  ],
}
`;
